dask_expr/_shuffle.py

```diff
@@ -716,6 +734,80 @@ def _simplify_up(self, parent):
             )
 
 
+class SortValues(BaseSetIndexSortValues):
+    _parameters = [
+        "frame",
+        "by",
+        "ascending",
+        "na_position",
+        "npartitions",
+        "partition_size",
+        "sort_function",
+        "sort_function_kwargs",
+    ]
+    _defaults = {
+        "partition_size": 128e6,
+        "ascending": True,
+        "npartitions": None,
+        "na_position": "last",
+        "sort_function": None,
+        "sort_function_kwargs": None,
+    }
+
+    @property
+    def sort_function(self):
+        if self.operand("sort_function") is not None:
+            return self.operand("sort_function")
+        return M.sort_values
+
+    @property
+    def sort_function_kwargs(self):
+        sort_kwargs = {
+            "by": self.by,
+            "ascending": self.ascending,
+            "na_position": self.na_position,
+        }
+        if self.operand("sort_function_kwargs") is not None:
+            sort_kwargs.update(self.operand("sort_function_kwargs"))
+        return sort_kwargs
+
+    @property
+    def _meta(self):
+        return self.frame._meta
+
+    def _lower(self):
+        by = self.frame[self.by[0]]
+        divisions, _, _, presorted = _calculate_divisions(
+            self.frame, by, self.npartitions, self.ascending
+        )
+        if presorted and self.npartitions == self.frame.npartitions:
+            return SortValuesBlockwise(
+                self.frame, self.sort_function, self.sort_function_kwargs
+            )
+
+        partitions = _SetPartitionsPreSetIndex(by, by._meta._constructor(divisions))
+        assigned = Assign(self.frame, "_partitions", partitions)
+        shuffled = Shuffle(
+            assigned,
+            "_partitions",
+            npartitions_out=len(divisions) - 1,
+            ignore_index=True,
+        )
+        return SortValuesBlockwise(
+            shuffled, self.sort_function, self.sort_function_kwargs
+        )
+
+    def _simplify_up(self, parent):
+        if isinstance(parent, Projection):
+            columns = parent.columns + self.by
```

Comment:
Couldn't this include the `by` column twice?

Reply:
Oh fair, will add a test

Reply:
Updated